astronomer/providers/google/cloud/triggers/gcs.py

```diff
@@ -165,3 +167,99 @@ async def _object_with_prefix_exists(
             bucket = client.get_bucket(bucket_name)
             object_response = await bucket.list_blobs(prefix=prefix)
             return object_response
+
+
+class GCSCheckBlobUpdateTimeTrigger(GCSBlobTrigger):
+    """
+    A trigger Makes an async call to GCS and check whether the object is updated in blob.
+
+    :param bucket: google cloud storage bucket name cloud storage where the objects are residing.
+    :param object_name: the file or folder present in the bucket
+    :param polling_period_seconds: polling period in seconds to check for file/folder
+    :param google_cloud_conn_id: reference to the Google Connection
+    :param hook_params: DIct object has delegate_to and impersonation_chain
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        object_name: str,
+        ts: datetime,
+        polling_period_seconds: float,
+        google_cloud_conn_id: str,
+        hook_params: dict,
+    ):
+        super().__init__()
+        self.bucket = bucket
+        self.object_name = object_name
+        self.ts = ts
+        self.polling_period_seconds = polling_period_seconds
+        self.google_cloud_conn_id: str = google_cloud_conn_id
+        self.hook_params = hook_params
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """
+        Serializes GCSCheckBlobUpdateTimeTrigger arguments and classpath.
+        """
+        return (
+            "astronomer.providers.google.cloud.triggers.gcs.GCSCheckBlobUpdateTimeTrigger",
+            {
+                "bucket": self.bucket,
+                "object_name": self.object_name,
+                "ts": self.ts,
+                "polling_period_seconds": self.polling_period_seconds,
+                "google_cloud_conn_id": self.google_cloud_conn_id,
+                "hook_params": self.hook_params,
+            },
+        )
+
+    async def run(self):
+        """
+        Simple loop until the relevant file/folder is found.
+        """
+        try:
+            hook = self._get_async_hook()
+            while True:
+                res = await self._is_blob_updated_after(
+                    hook=hook, bucket_name=self.bucket, object_name=self.object_name, ts=self.ts
+                )
+                if res:
+                    yield TriggerEvent({"status": "success", "message": res})
+                    return
+                await asyncio.sleep(self.polling_period_seconds)
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
+            return
+
+    def _get_async_hook(self) -> GCSHookAsync:
+        return GCSHookAsync(gcp_conn_id=self.google_cloud_conn_id, **self.hook_params)
+
+    async def _is_blob_updated_after(
```

Comment:
Any reason to keep `_is_blob_updated_after()` a private method on triggers rather than keeping it in hooks?

Reply:
Yes I this return the state of the blob object whether it is updated or not